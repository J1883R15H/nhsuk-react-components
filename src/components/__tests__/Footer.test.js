import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import Footer from '../footer';

describe('Footer', () => {
  it('matches snapshot', () => {
    const linkShallowMount = shallow(<Footer.Link>Hello</Footer.Link>);
    const footerShallowMount = shallow(
      <Footer>
        <Footer.Link>Hello</Footer.Link>
        <Footer.Copyright>MIT License</Footer.Copyright>
      </Footer>
    );
    expect(linkShallowMount).toMatchSnapshot();
    expect(footerShallowMount).toMatchSnapshot();
    linkShallowMount.unmount();
    footerShallowMount.unmount();
  });
  it('renders links properly', () => {
    const linkShallowMount = shallow(
      <Footer.Link href="test-url">Test Link</Footer.Link>
    );
    expect(
      linkShallowMount.containsMatchingElement(
        <a
          className="nhsuk-footer__list-item-link"
          target=""
          rel=""
          href="test-url"
        >
          Test Link
        </a>
      )
    ).toBeTruthy();
    linkShallowMount.unmount();
  });
  it('handles onclick', () => {
    const onClick = sinon.spy();
    const shallowMount = shallow(
      <Footer.Link onClick={onClick}>Error item</Footer.Link>
    );
    const shallowMountForAnonFunc = shallow(
      <Footer.Link>Error item</Footer.Link>
    );
    shallowMount.find('a').simulate('click');
    shallowMountForAnonFunc.find('a').simulate('click');
    sinon.assert.called(onClick);
    shallowMount.unmount();
  });
  it('renders copyright notice', () => {
    const shallowMount = shallow(
      <Footer.Copyright>MIT License</Footer.Copyright>
    );
    expect(shallowMount).toMatchSnapshot();
    expect(shallowMount.text()).toEqual('MIT License');
    expect(
      shallowMount.containsMatchingElement(
        <p className="nhsuk-footer__copyright"> MIT License</p>
      )
    );
    shallowMount.unmount();
  });
  it('adds the required attributes on openInNewWindow', () => {
    const linkShallowMount = shallow(
      <Footer.Link href="test-url" openInNewWindow>
        Test Link
      </Footer.Link>
    );
    expect(
      linkShallowMount.containsMatchingElement(
        <a
          className="nhsuk-footer__list-item-link"
          target="_blank"
          rel="noreferrer noopener"
          href="test-url"
        >
          Test Link
        </a>
      )
    ).toBeTruthy();
    linkShallowMount.unmount();
  });
  it('adds hidden text when supplied links', () => {
    const mountedFooterWithLinks = mount(
      <Footer>
        <Footer.Link>Test Link</Footer.Link>
        <Footer.Copyright>MIT License</Footer.Copyright>
      </Footer>
    );
    const mountedFooterWithoutLinks = mount(
      <Footer>
        <Footer.Copyright>MIT License</Footer.Copyright>
      </Footer>
    );
    expect(
      mountedFooterWithLinks.containsMatchingElement(
        <h2 className="nhsuk-u-visually-hidden">Support links</h2>
      )
    ).toBeTruthy();
    expect(
      mountedFooterWithoutLinks.containsMatchingElement(
        <h2 className="nhsuk-u-visually-hidden">Support links</h2>
      )
    ).toBeFalsy();
    mountedFooterWithLinks.unmount();
    mountedFooterWithoutLinks.unmount();
  });
});
