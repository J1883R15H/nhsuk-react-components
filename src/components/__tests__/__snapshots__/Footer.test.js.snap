// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Link
    className=""
    href="#"
    onClick={[Function]}
    openInNewWindow={false}
    style={Object {}}
>
    Hello
</Link>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="nhsuk-footer__list-item-link"
        href="#"
        onClick={[Function]}
        rel=""
        target=""
>
        Hello
</a>,
      "className": "nhsuk-footer__list-item ",
      "style": Object {},
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello",
        "className": "nhsuk-footer__list-item-link",
        "href": "#",
        "onClick": [Function],
        "rel": "",
        "target": "",
      },
      "ref": null,
      "rendered": "Hello",
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="nhsuk-footer__list-item-link"
          href="#"
          onClick={[Function]}
          rel=""
          target=""
>
          Hello
</a>,
        "className": "nhsuk-footer__list-item ",
        "style": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello",
          "className": "nhsuk-footer__list-item-link",
          "href": "#",
          "onClick": [Function],
          "rel": "",
          "target": "",
        },
        "ref": null,
        "rendered": "Hello",
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Footer matches snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    className=""
    id="nhsuk-footer"
    linksVisuallyHiddenText="Support links"
    role="contentinfo"
    style={Object {}}
>
    <Link
        className=""
        href="#"
        onClick={[Function]}
        openInNewWindow={false}
        style={Object {}}
    >
        Hello
    </Link>
    <Copyright>
        MIT License
    </Copyright>
</Footer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nhsuk-footer "
        id="nhsuk-footer"
        style={Object {}}
>
        <div
                className="nhsuk-width-container"
        >
                <div>
                        <h2
                                className="nhsuk-u-visually-hidden"
                        >
                                Support links
                        </h2>
                        <ul
                                className="nhsuk-footer__list"
                        >
                                <Link
                                        className=""
                                        href="#"
                                        onClick={[Function]}
                                        openInNewWindow={false}
                                        style={Object {}}
                                >
                                        Hello
                                </Link>
                        </ul>
                </div>
        </div>
</div>,
      "role": "contentinfo",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nhsuk-width-container"
>
          <div>
                    <h2
                              className="nhsuk-u-visually-hidden"
                    >
                              Support links
                    </h2>
                    <ul
                              className="nhsuk-footer__list"
                    >
                              <Link
                                        className=""
                                        href="#"
                                        onClick={[Function]}
                                        openInNewWindow={false}
                                        style={Object {}}
                              >
                                        Hello
                              </Link>
                    </ul>
          </div>
</div>,
        "className": "nhsuk-footer ",
        "id": "nhsuk-footer",
        "style": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h2
                            className="nhsuk-u-visually-hidden"
              >
                            Support links
              </h2>
              <ul
                            className="nhsuk-footer__list"
              >
                            <Link
                                          className=""
                                          href="#"
                                          onClick={[Function]}
                                          openInNewWindow={false}
                                          style={Object {}}
                            >
                                          Hello
                            </Link>
              </ul>
</div>,
            undefined,
          ],
          "className": "nhsuk-width-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="nhsuk-u-visually-hidden"
>
                  Support links
</h2>,
                <ul
                  className="nhsuk-footer__list"
>
                  <Link
                                    className=""
                                    href="#"
                                    onClick={[Function]}
                                    openInNewWindow={false}
                                    style={Object {}}
                  >
                                    Hello
                  </Link>
</ul>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Support links",
                  "className": "nhsuk-u-visually-hidden",
                },
                "ref": null,
                "rendered": "Support links",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Link
                      className=""
                      href="#"
                      onClick={[Function]}
                      openInNewWindow={false}
                      style={Object {}}
>
                      Hello
</Link>,
                  ],
                  "className": "nhsuk-footer__list",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Hello",
                      "className": "",
                      "href": "#",
                      "onClick": [Function],
                      "openInNewWindow": false,
                      "style": Object {},
                    },
                    "ref": null,
                    "rendered": "Hello",
                    "type": [Function],
                  },
                ],
                "type": "ul",
              },
            ],
            "type": "div",
          },
          undefined,
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nhsuk-footer "
          id="nhsuk-footer"
          style={Object {}}
>
          <div
                    className="nhsuk-width-container"
          >
                    <div>
                              <h2
                                        className="nhsuk-u-visually-hidden"
                              >
                                        Support links
                              </h2>
                              <ul
                                        className="nhsuk-footer__list"
                              >
                                        <Link
                                                  className=""
                                                  href="#"
                                                  onClick={[Function]}
                                                  openInNewWindow={false}
                                                  style={Object {}}
                                        >
                                                  Hello
                                        </Link>
                              </ul>
                    </div>
          </div>
</div>,
        "role": "contentinfo",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="nhsuk-width-container"
>
            <div>
                        <h2
                                    className="nhsuk-u-visually-hidden"
                        >
                                    Support links
                        </h2>
                        <ul
                                    className="nhsuk-footer__list"
                        >
                                    <Link
                                                className=""
                                                href="#"
                                                onClick={[Function]}
                                                openInNewWindow={false}
                                                style={Object {}}
                                    >
                                                Hello
                                    </Link>
                        </ul>
            </div>
</div>,
          "className": "nhsuk-footer ",
          "id": "nhsuk-footer",
          "style": Object {},
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h2
                                className="nhsuk-u-visually-hidden"
                >
                                Support links
                </h2>
                <ul
                                className="nhsuk-footer__list"
                >
                                <Link
                                                className=""
                                                href="#"
                                                onClick={[Function]}
                                                openInNewWindow={false}
                                                style={Object {}}
                                >
                                                Hello
                                </Link>
                </ul>
</div>,
              undefined,
            ],
            "className": "nhsuk-width-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="nhsuk-u-visually-hidden"
>
                    Support links
</h2>,
                  <ul
                    className="nhsuk-footer__list"
>
                    <Link
                                        className=""
                                        href="#"
                                        onClick={[Function]}
                                        openInNewWindow={false}
                                        style={Object {}}
                    >
                                        Hello
                    </Link>
</ul>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Support links",
                    "className": "nhsuk-u-visually-hidden",
                  },
                  "ref": null,
                  "rendered": "Support links",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Link
                        className=""
                        href="#"
                        onClick={[Function]}
                        openInNewWindow={false}
                        style={Object {}}
>
                        Hello
</Link>,
                    ],
                    "className": "nhsuk-footer__list",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "Hello",
                        "className": "",
                        "href": "#",
                        "onClick": [Function],
                        "openInNewWindow": false,
                        "style": Object {},
                      },
                      "ref": null,
                      "rendered": "Hello",
                      "type": [Function],
                    },
                  ],
                  "type": "ul",
                },
              ],
              "type": "div",
            },
            undefined,
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Footer renders copyright notice 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Copyright>
    MIT License
</Copyright>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "MIT License",
      "className": "nhsuk-footer__copyright",
    },
    "ref": null,
    "rendered": "MIT License",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "MIT License",
        "className": "nhsuk-footer__copyright",
      },
      "ref": null,
      "rendered": "MIT License",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
